### PR TITLE
[Feature-WIP](inverted index) step 1 for supporting range predicate pushing down to inverted index

### PR DESCRIPTION
**Pull request by airborne12**
# Proposed changes

Try to push down range predicate(like a>1 and a<5) to inverted index for better index filter performance.

## Problem summary

Describe your changes.

This is step 1 for "push down range predicate", in this pr, we introduced inverted index iterator evaluate for AndBlockColumnPredicate.

And we use col_id_to_predicates to construct per column predicates, 

We also refact apply_inverted_index in this pr.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 1]
```diff
@@ -642,52 +641,133 @@ bool SegmentIterator::_is_handle_predicate_by_fulltext(ColumnPredicate* predicat
...
+            if (res.code() == ErrorCode::NOT_IMPLEMENTED_ERROR) {
+                return Status::OK();
+            }
+            LOG(WARNING) << "failed to evaluate index"
+                         << ", column predicate type: range predicate"
+                         << ", error msg: " << res.code_as_string();
+            return res;
+        }
+    }
+    return Status::OK();
```
Comment (Tanya-W) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:742]: here use "return res" maybe better, then can remove redundant "return res" on line 730 and line 739

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 2]
```diff
@@ -642,52 +641,133 @@ bool SegmentIterator::_is_handle_predicate_by_fulltext(ColumnPredicate* predicat
...
+}
+
 Status SegmentIterator::_apply_inverted_index() {
     SCOPED_RAW_TIMER(&_opts.stats->inverted_index_filter_timer);
     size_t input_rows = _row_bitmap.cardinality();
     std::vector<ColumnPredicate*> remaining_predicates;
+    std::set<const ColumnPredicate*> no_need_to_pass_column_predicate_set;
```
Comment (Tanya-W) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:749]: can use a more clear name, such as "success_applied_column_predicate", or add some note.

**Review by Tanya-W — COMMENTED** [review]
LGTM

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.41 seconds
 stream load tsv:          472 seconds loaded 74807831229 Bytes, about 151 MB/s
 stream load json:         36 seconds loaded 2358488459 Bytes, about 62 MB/s
 stream load orc:          69 seconds loaded 1101869774 Bytes, about 15 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230210184851_clickbench_pr_94046.html

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**